src/applications/find-forms/actions/index.js

```diff
@@ -31,6 +35,39 @@ export const fetchFormsSuccess = (results, hasOnlyRetiredForms) => ({
   type: FETCH_FORMS_SUCCESS,
 });
 
+// =============
+// Update Results after forms is sorted
+// =============
+export const updateResults = results => ({
+  results,
+  type: UPDATE_RESULTS,
+});
+
+// =============
+// Update How To Sort
+// =============
+export const updateSortByPropertyName = sortByPropertyName => ({
+  sortByPropertyName,
+  type: UPDATE_HOW_TO_SORT,
+});
+
+export const updateSortByPropertyNameThunk = sortByPropertyName => (
+  dispatch,
+  getState,
+) => {
+  dispatch(updateSortByPropertyName(sortByPropertyName));
+
+  const clonedResults = cloneDeep(
+    getFindFormsAppState(getState()).results || [],
+  );
+
+  const sortedResults = clonedResults.sort((a, b) =>
+    sortTheResults(getFindFormsAppState(getState()).sortByPropertyName, a, b),
+  );
```

Comment:
Not necessary unless you want to, but sorting at the reducer might save on some complexity. It would make it a little easier here since your reducer has access to state by design. It would be something like 

```
// actions
export const updateSortByPropertyName = sortByPropertyName => ({
  sortByPropertyName,
  type: UPDATE_HOW_TO_SORT,
});

// reducer
// ...
case UPDATE_HOW_TO_SORT: 
    return {
        ...state,
         sortByPropertyName: action.sortByPropertyName,
         results: sortResults(action.sortByPropertyName, state.results)
    }
```

I think this is the more "Redux-y" way because the reducer does the calculations while the action describes the interaction as minimally as possible, but once Redux Thunks enter the equation it gets foggy since you can do `getState()` from the action creator.

Reply:
Ah this is a good point. That makes sense, a reducer reduces the two states to the needed one new state.